src/main/java/org/jenkinsci/plugins/GithubOAuthUserDetails.java

```diff
@@ -20,34 +19,8 @@ public class GithubOAuthUserDetails extends User implements UserDetails {
 
     private static final long serialVersionUID = 1L;
 
-    private boolean hasGrantedAuthorities;
-
-    private final GithubAuthenticationToken authenticationToken;
-
-    public GithubOAuthUserDetails(@NonNull String login, @NonNull GrantedAuthority[] authorities) {
+    public GithubOAuthUserDetails(@NonNull String login, @NonNull Collection<? extends GrantedAuthority> authorities) {
         super(login, "", true, true, true, true, authorities);
-        this.authenticationToken = null;
-        this.hasGrantedAuthorities = true;
     }
 
-    public GithubOAuthUserDetails(@NonNull String login, @NonNull GithubAuthenticationToken authenticationToken) {
-        super(login, "", true, true, true, true, new GrantedAuthority[0]);
-        this.authenticationToken = authenticationToken;
-        this.hasGrantedAuthorities = false;
-    }
-
-    @Override
-    public GrantedAuthority[] getAuthorities() {
-        if (!hasGrantedAuthorities) {
-            try {
-                GHUser user = authenticationToken.loadUser(getUsername());
-                if(user != null) {
-                    setAuthorities(authenticationToken.getAuthorities());
-                }
-            } catch (IOException e) {
-                throw new RuntimeException(e);
-            }
-        }
-        return super.getAuthorities();
-    }
```

Comment:
In Spring Security 6.x, the parent class is immutable, so pushing this logic up to the caller.